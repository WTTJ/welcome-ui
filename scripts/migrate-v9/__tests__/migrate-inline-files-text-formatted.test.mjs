import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - Text Components', () => {
  const tempDir = setupMigrationTest('inline-text-formatted-test', '__fixtures__')

  it('migrates Text components containing FormattedMessage to span with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'TextComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('text-component.tsx')
  })
})
