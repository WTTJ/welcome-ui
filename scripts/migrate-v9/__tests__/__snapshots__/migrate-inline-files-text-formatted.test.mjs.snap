// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Text Components > migrates Text components containing FormattedMessage to span with classNames > text-component.tsx 1`] = `
"import { Text } from 'welcome-ui/Text'

const FormattedMessage = ({
  defaultMessage,
  id,
}: {
  defaultMessage: string
  id: string
}): React.ReactNode => <span id={id}>{defaultMessage}</span>

export const TextComponent = () => {
  return (
    <div>
      <Text className="mt-md text-h3" as="h1" variant="h4">
        This is a paragraph with custom styling.
      </Text>
      <Text className="mt-auto text-center text-dark-50 text-sm" as="span">
        <FormattedMessage defaultMessage="Â© 2024 Acme Corp. All rights reserved." id="8XKXo+" />
      </Text>
    </div>
  )
}
"
`;
